{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from glob import glob\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "for f in glob('../assertions/*.new_archive/*gene*.csv'):\n",
    "\tnf = f.replace('gene', 'Gene')\n",
    "\tos.rename(f, nf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cp: Pathways.nodes.csv is not a directory\n",
      "cp: Sample.nodes.csv is not a directory\n",
      "cp: Pert.nodes.csv is not a directory\n",
      "cp: Protein.nodes.csv is not a directory\n",
      "cp: Perturbation.nodes.csv is not a directory\n",
      "cp: Samples.nodes.csv is not a directory\n",
      "cp: CRISPR.nodes.csv is not a directory\n",
      "cp: Protein.nodes.csv is not a directory\n",
      "cp: CNV.nodes.csv is not a directory\n",
      "cp: Modification.nodes.csv is not a directory\n",
      "cp: Expts.nodes.csv is not a directory\n",
      "cp: 2023.nodes.csv is not a directory\n",
      "cp: Protein.nodes.csv is not a directory\n",
      "cp: CNV.nodes.csv is not a directory\n",
      "cp: 2023.nodes.csv is not a directory\n",
      "cp: Pathways.nodes.csv is not a directory\n",
      "cp: Targets.nodes.csv is not a directory\n",
      "cp: Perturbation.nodes.csv is not a directory\n",
      "cp: KinomeScan.nodes.csv is not a directory\n",
      "cp: TextMining.nodes.csv is not a directory\n",
      "cp: Proteins.nodes.csv is not a directory\n",
      "cp: Type.nodes.csv is not a directory\n",
      "cp: Library.nodes.csv is not a directory\n",
      "cp: Expression.nodes.csv is not a directory\n",
      "cp: Mutations.nodes.csv is not a directory\n",
      "cp: Mouse.nodes.csv is not a directory\n",
      "cp: Binding.nodes.csv is not a directory\n",
      "cp: Expts.nodes.csv is not a directory\n",
      "cp: Signatures.nodes.csv is not a directory\n",
      "cp: Perturbation.nodes.csv is not a directory\n",
      "cp: Metabolites.nodes.csv is not a directory\n",
      "cp: Line.nodes.csv is not a directory\n",
      "cp: Seq.nodes.csv is not a directory\n",
      "cp: Expts.nodes.csv is not a directory\n",
      "cp: Kinase.nodes.csv is not a directory\n",
      "cp: NCI-60.nodes.csv is not a directory\n",
      "cp: Chemical.nodes.csv is not a directory\n",
      "cp: Curated.nodes.csv is not a directory\n",
      "cp: Human.nodes.csv is not a directory\n",
      "cp: Perturbation.nodes.csv is not a directory\n",
      "cp: Knockout.nodes.csv is not a directory\n",
      "cp: Mouse.nodes.csv is not a directory\n",
      "cp: Phenotype.nodes.csv is not a directory\n",
      "cp: Mutations.nodes.csv is not a directory\n",
      "cp: Methylation.nodes.csv is not a directory\n",
      "cp: CNV.nodes.csv is not a directory\n",
      "cp: Navigator.nodes.csv is not a directory\n",
      "cp: Virus.nodes.csv is not a directory\n",
      "cp: Perturbation.nodes.csv is not a directory\n",
      "cp: Protein.nodes.csv is not a directory\n",
      "cp: Human.nodes.csv is not a directory\n",
      "cp: Level.nodes.csv is not a directory\n",
      "cp: 2023.nodes.csv is not a directory\n",
      "cp: Expression.nodes.csv is not a directory\n",
      "cp: Signatures.nodes.csv is not a directory\n",
      "cp: Perturbation.nodes.csv is not a directory\n",
      "cp: 2022.nodes.csv is not a directory\n",
      "cp: Microarray.nodes.csv is not a directory\n",
      "cp: mRNA.nodes.csv is not a directory\n",
      "cp: Dependency.nodes.csv is not a directory\n",
      "cp: PWMs.nodes.csv is not a directory\n",
      "cp: Curated.nodes.csv is not a directory\n",
      "cp: Disease.nodes.csv is not a directory\n",
      "cp: mRNA.nodes.csv is not a directory\n",
      "cp: Human.nodes.csv is not a directory\n",
      "cp: DB.nodes.csv is not a directory\n",
      "cp: Catalog.nodes.csv is not a directory\n",
      "cp: Map.nodes.csv is not a directory\n",
      "cp: Targets.nodes.csv is not a directory\n",
      "cp: Proteomics.nodes.csv is not a directory\n",
      "cp: Modification.nodes.csv is not a directory\n",
      "cp: Pathways.nodes.csv is not a directory\n",
      "cp: Phenotype.nodes.csv is not a directory\n",
      "cp: Textmining.nodes.csv is not a directory\n",
      "cp: Chemical.nodes.csv is not a directory\n",
      "cp: Sapiens.nodes.csv is not a directory\n",
      "cp: Perturbation.nodes.csv is not a directory\n",
      "cp: Signatures.nodes.csv is not a directory\n",
      "cp: Mining.nodes.csv is not a directory\n",
      "cp: 2023.nodes.csv is not a directory\n",
      "cp: eQTL.nodes.csv is not a directory\n",
      "cp: Perturbations.nodes.csv is not a directory\n",
      "cp: PFOCR.nodes.csv is not a directory\n",
      "cp: Mutations.nodes.csv is not a directory\n",
      "cp: Lines.nodes.csv is not a directory\n",
      "cp: Disease.nodes.csv is not a directory\n",
      "cp: KiNativ.nodes.csv is not a directory\n",
      "cp: Protein.nodes.csv is not a directory\n",
      "cp: Perturbation.nodes.csv is not a directory\n",
      "cp: Conserved.nodes.csv is not a directory\n",
      "cp: Curated.nodes.csv is not a directory\n"
     ]
    }
   ],
   "source": [
    "pattern = '(?P<prefix>.*)/(?P<node_type>.*).nodes.csv'\n",
    "unassigned = True\n",
    "gene_df = None\n",
    "for f in glob('../assertions/*.new_archive/*.nodes.csv'):\n",
    "\tnode_type = re.match(pattern, f).groupdict()['node_type']\n",
    "\tif node_type == \"Gene\":\n",
    "\t\tdf = pd.read_csv(f, index_col=0)\n",
    "\t\tif unassigned:\n",
    "\t\t\tgene_df = df\n",
    "\t\t\tunassigned = False\n",
    "\t\telse:\n",
    "\t\t\tgene_df =pd.concat([gene_df, df])\n",
    "\telse:\n",
    "\t\tos.system('cp %s ../assertions/all/%s.nodes.csv'%(f, node_type))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'../assertions/jensencompartmentcurated.new_archive/COMPARTMENTS Curated.nodes.csv'"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>label</th>\n",
       "      <th>namespace</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5578</th>\n",
       "      <td>PRKCA</td>\n",
       "      <td>NCBI Entrez</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5581</th>\n",
       "      <td>PRKCE</td>\n",
       "      <td>NCBI Entrez</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5588</th>\n",
       "      <td>PRKCQ</td>\n",
       "      <td>NCBI Entrez</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>NCBI Entrez</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5170</th>\n",
       "      <td>PDPK1</td>\n",
       "      <td>NCBI Entrez</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28901</th>\n",
       "      <td>IGKV1D-16</td>\n",
       "      <td>NCBI Entrez</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28915</th>\n",
       "      <td>IGKV3-7</td>\n",
       "      <td>NCBI Entrez</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616</th>\n",
       "      <td>BCRP4</td>\n",
       "      <td>NCBI Entrez</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100384884</th>\n",
       "      <td>IGHV1OR15-5</td>\n",
       "      <td>NCBI Entrez</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28317</th>\n",
       "      <td>IGHV4OR15-8</td>\n",
       "      <td>NCBI Entrez</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31432 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 label    namespace  type\n",
       "id                                       \n",
       "5578             PRKCA  NCBI Entrez  gene\n",
       "5581             PRKCE  NCBI Entrez  gene\n",
       "5588             PRKCQ  NCBI Entrez  gene\n",
       "207               AKT1  NCBI Entrez  gene\n",
       "5170             PDPK1  NCBI Entrez  gene\n",
       "...                ...          ...   ...\n",
       "28901        IGKV1D-16  NCBI Entrez  gene\n",
       "28915          IGKV3-7  NCBI Entrez  gene\n",
       "616              BCRP4  NCBI Entrez  gene\n",
       "100384884  IGHV1OR15-5  NCBI Entrez  gene\n",
       "28317      IGHV4OR15-8  NCBI Entrez  gene\n",
       "\n",
       "[31432 rows x 3 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_df[~gene_df.index.duplicated(keep='first')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import time\n",
    "from tqdm import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = \"https://maayanlab.cloud/Harmonizome/api/1.0/gene_set?cursor=%d\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 10%|▉         | 448/4647 [02:59<27:59,  2.50it/s]\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[14], line 4\u001b[0m\n\u001b[1;32m      2\u001b[0m cursor \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0\u001b[39m\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m cursor \u001b[38;5;129;01min\u001b[39;00m tqdm(\u001b[38;5;28mrange\u001b[39m(\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m464610\u001b[39m, \u001b[38;5;241m100\u001b[39m)):\n\u001b[0;32m----> 4\u001b[0m \tres \u001b[38;5;241m=\u001b[39m \u001b[43mrequests\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m%\u001b[39;49m\u001b[43mcursor\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      5\u001b[0m \t\u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m res\u001b[38;5;241m.\u001b[39mok:\n\u001b[1;32m      6\u001b[0m \t\t\u001b[38;5;28mprint\u001b[39m(res\u001b[38;5;241m.\u001b[39mtext)\n",
      "File \u001b[0;32m~/Documents/codebase/HarmonizomeKG/venv/lib/python3.12/site-packages/requests/api.py:73\u001b[0m, in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     62\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget\u001b[39m(url, params\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m     63\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Sends a GET request.\u001b[39;00m\n\u001b[1;32m     64\u001b[0m \n\u001b[1;32m     65\u001b[0m \u001b[38;5;124;03m    :param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[38;5;124;03m    :rtype: requests.Response\u001b[39;00m\n\u001b[1;32m     71\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m---> 73\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mget\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mparams\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mparams\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/Documents/codebase/HarmonizomeKG/venv/lib/python3.12/site-packages/requests/api.py:59\u001b[0m, in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     55\u001b[0m \u001b[38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we\u001b[39;00m\n\u001b[1;32m     56\u001b[0m \u001b[38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001b[39;00m\n\u001b[1;32m     57\u001b[0m \u001b[38;5;66;03m# cases, and look like a memory leak in others.\u001b[39;00m\n\u001b[1;32m     58\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m sessions\u001b[38;5;241m.\u001b[39mSession() \u001b[38;5;28;01mas\u001b[39;00m session:\n\u001b[0;32m---> 59\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43msession\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/Documents/codebase/HarmonizomeKG/venv/lib/python3.12/site-packages/requests/sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    584\u001b[0m send_kwargs \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m    585\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtimeout\u001b[39m\u001b[38;5;124m\"\u001b[39m: timeout,\n\u001b[1;32m    586\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m: allow_redirects,\n\u001b[1;32m    587\u001b[0m }\n\u001b[1;32m    588\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[0;32m--> 589\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mprep\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43msend_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    591\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[0;32m~/Documents/codebase/HarmonizomeKG/venv/lib/python3.12/site-packages/requests/sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    700\u001b[0m start \u001b[38;5;241m=\u001b[39m preferred_clock()\n\u001b[1;32m    702\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[0;32m--> 703\u001b[0m r \u001b[38;5;241m=\u001b[39m \u001b[43madapter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    705\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n\u001b[1;32m    706\u001b[0m elapsed \u001b[38;5;241m=\u001b[39m preferred_clock() \u001b[38;5;241m-\u001b[39m start\n",
      "File \u001b[0;32m~/Documents/codebase/HarmonizomeKG/venv/lib/python3.12/site-packages/requests/adapters.py:667\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    664\u001b[0m     timeout \u001b[38;5;241m=\u001b[39m TimeoutSauce(connect\u001b[38;5;241m=\u001b[39mtimeout, read\u001b[38;5;241m=\u001b[39mtimeout)\n\u001b[1;32m    666\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 667\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43murlopen\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    668\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    669\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    670\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    671\u001b[0m \u001b[43m        \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    672\u001b[0m \u001b[43m        \u001b[49m\u001b[43mredirect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    673\u001b[0m \u001b[43m        \u001b[49m\u001b[43massert_same_host\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    674\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    675\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    676\u001b[0m \u001b[43m        \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmax_retries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    677\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    678\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    679\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    681\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m    682\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(err, request\u001b[38;5;241m=\u001b[39mrequest)\n",
      "File \u001b[0;32m~/Documents/codebase/HarmonizomeKG/venv/lib/python3.12/site-packages/urllib3/connectionpool.py:789\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    786\u001b[0m response_conn \u001b[38;5;241m=\u001b[39m conn \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m release_conn \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m    788\u001b[0m \u001b[38;5;66;03m# Make the request on the HTTPConnection object\u001b[39;00m\n\u001b[0;32m--> 789\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    790\u001b[0m \u001b[43m    \u001b[49m\u001b[43mconn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    791\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    792\u001b[0m \u001b[43m    \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    793\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout_obj\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    794\u001b[0m \u001b[43m    \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    795\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    796\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    797\u001b[0m \u001b[43m    \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mretries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    798\u001b[0m \u001b[43m    \u001b[49m\u001b[43mresponse_conn\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mresponse_conn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    799\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpreload_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    800\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdecode_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    801\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kw\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    802\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    804\u001b[0m \u001b[38;5;66;03m# Everything went great!\u001b[39;00m\n\u001b[1;32m    805\u001b[0m clean_exit \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n",
      "File \u001b[0;32m~/Documents/codebase/HarmonizomeKG/venv/lib/python3.12/site-packages/urllib3/connectionpool.py:536\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    534\u001b[0m \u001b[38;5;66;03m# Receive the response from the server\u001b[39;00m\n\u001b[1;32m    535\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 536\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mgetresponse\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    537\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (BaseSSLError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[1;32m    538\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_raise_timeout(err\u001b[38;5;241m=\u001b[39me, url\u001b[38;5;241m=\u001b[39murl, timeout_value\u001b[38;5;241m=\u001b[39mread_timeout)\n",
      "File \u001b[0;32m~/Documents/codebase/HarmonizomeKG/venv/lib/python3.12/site-packages/urllib3/connection.py:464\u001b[0m, in \u001b[0;36mHTTPConnection.getresponse\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    461\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mresponse\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m HTTPResponse\n\u001b[1;32m    463\u001b[0m \u001b[38;5;66;03m# Get the response from http.client.HTTPConnection\u001b[39;00m\n\u001b[0;32m--> 464\u001b[0m httplib_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43msuper\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mgetresponse\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    466\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    467\u001b[0m     assert_header_parsing(httplib_response\u001b[38;5;241m.\u001b[39mmsg)\n",
      "File \u001b[0;32m/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/http/client.py:1423\u001b[0m, in \u001b[0;36mHTTPConnection.getresponse\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1421\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m   1422\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 1423\u001b[0m         \u001b[43mresponse\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbegin\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1424\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m:\n\u001b[1;32m   1425\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mclose()\n",
      "File \u001b[0;32m/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/http/client.py:331\u001b[0m, in \u001b[0;36mHTTPResponse.begin\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    329\u001b[0m \u001b[38;5;66;03m# read until we get a non-100 response\u001b[39;00m\n\u001b[1;32m    330\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[0;32m--> 331\u001b[0m     version, status, reason \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_read_status\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    332\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m status \u001b[38;5;241m!=\u001b[39m CONTINUE:\n\u001b[1;32m    333\u001b[0m         \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[0;32m/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/http/client.py:292\u001b[0m, in \u001b[0;36mHTTPResponse._read_status\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    291\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_read_status\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[0;32m--> 292\u001b[0m     line \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mstr\u001b[39m(\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mreadline\u001b[49m\u001b[43m(\u001b[49m\u001b[43m_MAXLINE\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m)\u001b[49m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124miso-8859-1\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    293\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(line) \u001b[38;5;241m>\u001b[39m _MAXLINE:\n\u001b[1;32m    294\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m LineTooLong(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mstatus line\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socket.py:707\u001b[0m, in \u001b[0;36mSocketIO.readinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    705\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[1;32m    706\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 707\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sock\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrecv_into\u001b[49m\u001b[43m(\u001b[49m\u001b[43mb\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    708\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m timeout:\n\u001b[1;32m    709\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_timeout_occurred \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n",
      "File \u001b[0;32m/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/ssl.py:1252\u001b[0m, in \u001b[0;36mSSLSocket.recv_into\u001b[0;34m(self, buffer, nbytes, flags)\u001b[0m\n\u001b[1;32m   1248\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m flags \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m   1249\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[1;32m   1250\u001b[0m           \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnon-zero flags not allowed in calls to recv_into() on \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m\n\u001b[1;32m   1251\u001b[0m           \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__class__\u001b[39m)\n\u001b[0;32m-> 1252\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43mnbytes\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbuffer\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1253\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1254\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28msuper\u001b[39m()\u001b[38;5;241m.\u001b[39mrecv_into(buffer, nbytes, flags)\n",
      "File \u001b[0;32m/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/ssl.py:1104\u001b[0m, in \u001b[0;36mSSLSocket.read\u001b[0;34m(self, len, buffer)\u001b[0m\n\u001b[1;32m   1102\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m   1103\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m buffer \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m-> 1104\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sslobj\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mlen\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbuffer\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1105\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1106\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sslobj\u001b[38;5;241m.\u001b[39mread(\u001b[38;5;28mlen\u001b[39m)\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "links = {}\n",
    "cursor = 0\n",
    "for cursor in tqdm(range(0, 464610, 100)):\n",
    "\tres = requests.get(url%cursor)\n",
    "\tif not res.ok:\n",
    "\t\tprint(res.text)\n",
    "\t\tbreak\n",
    "\tfor i in res.json()[\"entities\"]:\n",
    "\t\ts = i[\"name\"].split(\"/\")\n",
    "\t\tdataset = s[-1]\n",
    "\t\tname = \"/\".join(s[:-1])\n",
    "\t\tif dataset not in links:\n",
    "\t\t\tlinks[dataset] = set()\n",
    "\t\tlinks[dataset].add(name)\n",
    "\ttime.sleep(0.1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Infralimbic area, layer 2', '3']"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "i[\"name\"].split(\"/\")[:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KEGG Pathways downloaded.\n"
     ]
    }
   ],
   "source": [
    "\"\"\"---------------------------------------------------------------------------\n",
    "A utility script for downloading data from the Harmonizome, with the ability\n",
    "to configure which datasets and which download types from which to download.\n",
    "Note that all content decompressed is roughly 30GB. The default is to not\n",
    "decompress the files on download.\n",
    "Dependencies:\n",
    "- requests is an HTTP library with an easy-to-use API:\n",
    "  http://docs.python-requests.org/en/latest/\n",
    "---------------------------------------------------------------------------\"\"\"\n",
    "\n",
    "\n",
    "import requests\n",
    "import os\n",
    "import zlib\n",
    "\n",
    "\n",
    "def _download_file(response, filename):\n",
    "    with open(filename, 'wb') as f:\n",
    "        for chunk in response.iter_content(chunk_size=1024):\n",
    "            f.write(chunk)\n",
    "\n",
    "\n",
    "def _download_and_decompress_file(response, filename):\n",
    "    decompressor = zlib.decompressobj(16 + zlib.MAX_WBITS)\n",
    "    filename = filename[:-3]\n",
    "    with open(filename, 'w+') as f:\n",
    "        while True:\n",
    "            chunk = response.raw.read(1024)\n",
    "            if not chunk:\n",
    "                break\n",
    "            string = decompressor.decompress(chunk)\n",
    "            f.write(string.decode('ascii'))\n",
    "\n",
    "\n",
    "def download_datasets(selected_datasets, selected_downloads, decompress=False):\n",
    "    for dataset, path in selected_datasets:\n",
    "        if not os.path.exists(\"out/\"+dataset):\n",
    "            os.mkdir(\"out/\"+dataset)\n",
    "\n",
    "        for downloadable in selected_downloads:\n",
    "            url = 'https://maayanlab.cloud/static/hdfs/harmonizome/data/%s/%s' %\\\n",
    "                  (path, downloadable)\n",
    "            response = requests.get(url, stream=True)\n",
    "            filename = 'out/%s/%s' % (dataset, downloadable)\n",
    "\n",
    "            # Not every dataset has all downloadables.\n",
    "            if response.status_code != 200:\n",
    "                continue\n",
    "\n",
    "            if decompress and 'txt.gz' in filename:\n",
    "                _download_and_decompress_file(response, filename)\n",
    "            else:\n",
    "                _download_file(response, filename)\n",
    "\n",
    "        print('%s downloaded.' % dataset)\n",
    "\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    # Uncomment a dataset to download it.\n",
    "    download_datasets([\n",
    "        ('Achilles Cell Line Gene Essentiality Profiles', 'achilles'),\n",
    "        ('Allen Brain Atlas Adult Human Brain Tissue Gene Expression Profiles', 'brainatlasadulthuman'),\n",
    "        ('Allen Brain Atlas Adult Mouse Brain Tissue Gene Expression Profiles', 'brainatlasadultmouse'),\n",
    "        ('Allen Brain Atlas Developing Human Brain Tissue Gene Expression Profiles by Microarray', 'brainatlasdevelopmentalhumanmicroarray'),\n",
    "        ('Allen Brain Atlas Developing Human Brain Tissue Gene Expression Profiles by RNA-seq', 'brainatlasdevelopmentalhumanrnaseq'),\n",
    "        ('Allen Brain Atlas Prenatal Human Brain Tissue Gene Expression Profiles', 'brainatlasprenatalhuman'),\n",
    "        ('BIND Biomolecular Interactions', 'bind'),\n",
    "        ('BioGPS Cell Line Gene Expression Profiles', 'biogpsnci60'),\n",
    "        ('BioGPS Human Cell Type and Tissue Gene Expression Profiles', 'biogpshuman'),\n",
    "        ('BioGPS Mouse Cell Type and Tissue Gene Expression Profiles', 'biogpsmouse'),\n",
    "        ('BioGRID Protein-Protein Interactions', 'biogrid'),\n",
    "        ('Biocarta Pathways', 'biocarta'),\n",
    "        ('CCLE Cell Line Gene CNV Profiles', 'cclecnv'),\n",
    "        ('CCLE Cell Line Gene Expression Profiles', 'cclemrna'),\n",
    "        ('CCLE Cell Line Gene Mutation Profiles', 'cclemut'),\n",
    "        ('CHEA Transcription Factor Binding Site Profiles', 'chea'),\n",
    "        ('CHEA Transcription Factor Targets', 'cheappi'),\n",
    "        ('CMAP Signatures of Differentially Expressed Genes for Small Molecules', 'cmap'),\n",
    "        ('COMPARTMENTS Curated Protein Localization Evidence Scores', 'jensencompartmentcurated'),\n",
    "        ('COMPARTMENTS Experimental Protein Localization Evidence Scores', 'jensencompartmentexpts'),\n",
    "        ('COMPARTMENTS Text-mining Protein Localization Evidence Scores', 'jensencompartmenttextmining'),\n",
    "        ('CORUM Protein Complexes', 'corum'),\n",
    "        ('COSMIC Cell Line Gene CNV Profiles', 'cosmiccnv'),\n",
    "        ('COSMIC Cell Line Gene Mutation Profiles', 'cosmicmut'),\n",
    "        ('CTD Gene-Chemical Interactions', 'ctdchemical'),\n",
    "        ('CTD Gene-Disease Associations', 'ctddisease'),\n",
    "        ('ClinVar SNP-Phenotype Associations', 'clinvar'),\n",
    "        ('Combined Pathways Pathways', 'combinedpathways'),\n",
    "        ('dbGAP Gene-Trait Associations', 'dbgap'),\n",
    "        ('DEPOD Substrates of Phosphatases', 'depod'),\n",
    "        ('DIP Protein-Protein Interactions', 'dip'),\n",
    "        ('DISEASES Curated Gene-Disease Assocation Evidence Scores', 'jensendiseasecurated'),\n",
    "        ('DISEASES Experimental Gene-Disease Assocation Evidence Scores', 'jensendiseaseexpts'),\n",
    "        ('DISEASES Text-mining Gene-Disease Assocation Evidence Scores', 'jensendiseasetextmining'),\n",
    "        ('DrugBank Drug Targets', 'drugbank'),\n",
    "        ('ENCODE Histone Modification Site Profiles', 'encodehm'),\n",
    "        ('ENCODE Transcription Factor Binding Site Profiles', 'encodetf'),\n",
    "        ('ENCODE Transcription Factor Targets', 'encodetfppi'),\n",
    "        ('ESCAPE Omics Signatures of Genes and Proteins for Stem Cells', 'escape'),\n",
    "        ('GAD Gene-Disease Associations', 'gad'),\n",
    "        ('GAD High Level Gene-Disease Associations', 'gadhighlevel'),\n",
    "        ('GDSC Cell Line Gene Expression Profiles', 'gdsc'),\n",
    "        ('GEO Signatures of Differentially Expressed Genes for Diseases', 'geodisease'),\n",
    "        ('GEO Signatures of Differentially Expressed Genes for Gene Perturbations', 'geogene'),\n",
    "        ('GEO Signatures of Differentially Expressed Genes for Kinase Perturbations', 'geokinase'),\n",
    "        ('GEO Signatures of Differentially Expressed Genes for Small Molecules', 'geochemical'),\n",
    "        ('GEO Signatures of Differentially Expressed Genes for Transcription Factor Perturbations', 'geotf'),\n",
    "        ('GEO Signatures of Differentially Expressed Genes for Viral Infections', 'geovirus'),\n",
    "        ('GO Biological Process Annotations', 'gobp'),\n",
    "        ('GO Cellular Component Annotations', 'gocc'),\n",
    "        ('GO Molecular Function Annotations', 'gomf'),\n",
    "        ('GTEx Tissue Gene Expression Profiles', 'gtextissue'),\n",
    "        ('GTEx Tissue Sample Gene Expression Profiles', 'gtexsample'),\n",
    "        ('GTEx eQTL', 'gtexeqtl'),\n",
    "        ('GWAS Catalog SNP-Phenotype Associations', 'gwascatalog'),\n",
    "        ('GWASdb SNP-Disease Associations', 'gwasdbdisease'),\n",
    "        ('GWASdb SNP-Phenotype Associations', 'gwasdbphenotype'),\n",
    "        ('GeneRIF Biological Term Annotations', 'generif'),\n",
    "        ('GeneSigDB Published Gene Signatures', 'genesigdb'),\n",
    "        ('Graph of Medicine EHR Text-mining Clinical Term Annotations', 'graphofmedicine'),\n",
    "        ('Guide to Pharmacology Chemical Ligands of Receptors', 'guidetopharmchemical'),\n",
    "        ('Guide to Pharmacology Protein Ligands of Receptors', 'guidetopharmprotein'),\n",
    "        ('HMDB Metabolites of Enzymes', 'hmdb'),\n",
    "        ('HPA Cell Line Gene Expression Profiles', 'hpacelllines'),\n",
    "        ('HPA Tissue Gene Expression Profiles', 'hpatissuesmrna'),\n",
    "        ('HPA Tissue Protein Expression Profiles', 'hpatissuesprotein'),\n",
    "        ('HPA Tissue Sample Gene Expression Profiles', 'hpasamples'),\n",
    "        ('HPM Cell Type and Tissue Protein Expression Profiles', 'hpm'),\n",
    "        ('HPO Gene-Disease Associations', 'hpo'),\n",
    "        ('HPRD Protein-Protein Interactions', 'hprd'),\n",
    "        ('Heiser et al., PNAS, 2011 Cell Line Gene Expression Profiles', 'heiser'),\n",
    "        ('HuGE Navigator Gene-Phenotype Associations', 'hugenavigator'),\n",
    "        ('Hub Proteins Protein-Protein Interactions', 'hubs'),\n",
    "        ('HumanCyc Biomolecular Interactions', 'humancycppi'),\n",
    "        ('HumanCyc Pathways', 'humancyc'),\n",
    "        ('IntAct Biomolecular Interactions', 'intact'),\n",
    "        ('InterPro Predicted Protein Domain Annotations', 'interpro'),\n",
    "        ('JASPAR Predicted Transcription Factor Targets', 'jasparpwm'),\n",
    "        ('KEA Substrates of Kinases', 'kea'),\n",
    "        ('KEGG Biomolecular Interactions', 'keggppi'),\n",
    "        ('KEGG Pathways', 'kegg'),\n",
    "        ('Kinativ Kinase Inhibitor Bioactivity Profiles', 'kinativ'),\n",
    "        ('KinomeScan Kinase Inhibitor Targets', 'kinomescan'),\n",
    "        ('Klijn et al., Nat. Biotechnol., 2015 Cell Line Gene CNV Profiles', 'klijncnv'),\n",
    "        ('Klijn et al., Nat. Biotechnol., 2015 Cell Line Gene Expression Profiles', 'klijnmrna'),\n",
    "        ('Klijn et al., Nat. Biotechnol., 2015 Cell Line Gene Mutation Profiles', 'klijnmut'),\n",
    "        ('LINCS L1000 CMAP Signatures of Differentially Expressed Genes for Gene Knockdowns', 'lincscmapgene'),\n",
    "        ('LINCS L1000 CMAP Signatures of Differentially Expressed Genes for Small Molecules', 'lincscmapchemical'),\n",
    "        ('LOCATE Curated Protein Localization Annotations', 'locate'),\n",
    "        ('LOCATE Predicted Protein Localization Annotations', 'locatepredicted'),\n",
    "        ('MPO Gene-Phenotype Associations', 'mgimpo'),\n",
    "        ('MSigDB Cancer Gene Co-expression Modules', 'msigdbcomp'),\n",
    "        ('MSigDB Signatures of Differentially Expressed Genes for Cancer Gene Perturbations', 'msigdbonc'),\n",
    "        ('MiRTarBase microRNA Targets', 'mirtarbase'),\n",
    "        ('MotifMap Predicted Transcription Factor Targets', 'motifmap'),\n",
    "        ('NURSA Protein Complexes', 'nursa'),\n",
    "        ('NURSA Protein-Protein Interactions', 'nursappi'),\n",
    "        ('OMIM Gene-Disease Associations', 'omim'),\n",
    "        ('PANTHER Biomolecular Interactions', 'pantherppi'),\n",
    "        ('PANTHER Pathways', 'panther'),\n",
    "        ('PID Biomolecular Interactions', 'pidppi'),\n",
    "        ('PID Pathways', 'pid'),\n",
    "        ('Pathway Commons Protein-Protein Interactions', 'pc'),\n",
    "        ('PhosphoSitePlus Phosphosite-Disease Associations', 'phosphositeplusdisease'),\n",
    "        ('PhosphoSitePlus Substrates of Kinases', 'phosphositeplus'),\n",
    "        ('Phosphosite Textmining Biological Term Annotations', 'phosphositetextmining'),\n",
    "        ('ProteomicsDB Cell Type and Tissue Protein Expression Profiles', 'proteomicsdb'),\n",
    "        ('Reactome Biomolecular Interactions', 'reactomeppi'),\n",
    "        ('Reactome Pathways', 'reactome'),\n",
    "        ('Recon X Predicted Biomolecular Interactions', 'reconx'),\n",
    "        ('Roadmap Epigenomics Cell and Tissue DNA Accessibility Profiles', 'epigenomicsdnaaccessibility'),\n",
    "        ('Roadmap Epigenomics Cell and Tissue DNA Methylation Profiles', 'epigenomicsdnamethylation'),\n",
    "        ('Roadmap Epigenomics Cell and Tissue Gene Expression Profiles', 'epigenomicsmrna'),\n",
    "        ('Roadmap Epigenomics Histone Modification Site Profiles', 'epigenomicshm'),\n",
    "        ('SILAC Phosphoproteomics Signatures of Differentially Phosphorylated Proteins for Drugs', 'silacdrug'),\n",
    "        ('SILAC Phosphoproteomics Signatures of Differentially Phosphorylated Proteins for Gene Perturbations', 'silacgene'),\n",
    "        ('SILAC Phosphoproteomics Signatures of Differentially Phosphorylated Proteins for Protein Ligands', 'silacligand'),\n",
    "        ('SNPedia SNP-Phenotype Associations', 'snpedia'),\n",
    "        ('TCGA Signatures of Differentially Expressed Genes for Tumors', 'tcga'),\n",
    "        ('TISSUES Curated Tissue Protein Expression Evidence Scores', 'jensentissuecurated'),\n",
    "        ('TISSUES Experimental Tissue Protein Expression Evidence Scores', 'jensentissueexpts'),\n",
    "        ('TISSUES Text-mining Tissue Protein Expression Evidence Scores', 'jensentissuetextmining'),\n",
    "        ('TRANSFAC Curated Transcription Factor Targets', 'transfac'),\n",
    "        ('TRANSFAC Predicted Transcription Factor Targets', 'transfacpwm'),\n",
    "        ('TargetScan Predicted Conserved microRNA Targets', 'targetscan'),\n",
    "        ('TargetScan Predicted Nonconserved microRNA Targets', 'targetscannonconserved'),\n",
    "        ('Virus MINT Protein-Viral Protein Interactions', 'virusmintppi'),\n",
    "        ('Virus MINT Protein-Virus Interactions', 'virusmint'),\n",
    "        ('Wikipathways Pathways', 'wikipathways'),\n",
    "    ], [\n",
    "        # 'gene_attribute_matrix.txt.gz',\n",
    "        # 'gene_attribute_edges.txt.gz',\n",
    "        # 'gene_set_library_crisp.txt.gz',\n",
    "        # 'gene_set_library_up_crisp.txt.gz',\n",
    "        # 'gene_set_library_dn_crisp.txt.gz',\n",
    "        # 'attribute_set_library_crisp.txt.gz',\n",
    "        # 'attribute_set_library_up_crisp.txt.gz',\n",
    "        # 'attribute_set_library_dn_crisp.txt.gz',\n",
    "        # 'gene_similarity_matrix_cosine.txt.gz',\n",
    "        # 'attribute_similarity_matrix_cosine.txt.gz',\n",
    "        # 'gene_list_terms.txt.gz',\n",
    "        'attribute_list_entries.txt.gz',\n",
    "        # 'processing_script.m'\n",
    "    ], decompress=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
